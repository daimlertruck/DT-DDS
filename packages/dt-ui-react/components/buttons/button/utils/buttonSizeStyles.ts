import { CustomTheme as Theme } from '../../../../types';
import { ButtonSizes } from '../constants/types';

export const buttonSizeStyles = (size: ButtonSizes, theme: Theme) =>
  ({
    large: `
      ${theme.fontStyles.button1}
      padding: ${theme.spacing['2xs']} ${theme.spacing.s};
    `,
    medium: `
      ${theme.fontStyles.button2}
      padding: ${theme.spacing['3xs']} ${theme.spacing.xs};
    `,
    small: `
      ${theme.fontStyles.button2}
      padding: ${theme.spacing['4xs']} ${theme.spacing['2xs']};
    `,
  }[size]);
