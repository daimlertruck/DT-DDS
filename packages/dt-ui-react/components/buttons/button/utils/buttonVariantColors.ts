import { CustomTheme as Theme } from '../../../../types';
import {
  ButtonVariant,
  ContainedColors,
  OutlinedColors,
  TextColors,
} from '../constants/types';

interface ColorKeys {
  color: string;
  bgColor: string;
  hoverBgColor?: string;
  hoverColor?: string;
}

export const buttonVariantColors = (
  color: ContainedColors | OutlinedColors,
  theme: Theme
): Record<ButtonVariant, ColorKeys> => ({
  contained: {
    color: theme.colors.alpha.white,
    bgColor: theme.palette.alpha[color],
    hoverBgColor: theme.palette.alpha[`${color}Dark_300` as ContainedColors],
  },
  outlined: {
    color:
      color === 'neutral'
        ? theme.colors.alpha.gray_200
        : theme.palette.alpha[color],
    bgColor: theme.colors.alpha.white,
    hoverBgColor: theme.palette.alpha[`${color}Light_300` as OutlinedColors],
  },
  text: {
    color: theme.palette.alpha[color],
    bgColor: 'transparent',
    hoverColor: theme.palette.alpha[`${color as TextColors}Dark_300`],
  },
});
