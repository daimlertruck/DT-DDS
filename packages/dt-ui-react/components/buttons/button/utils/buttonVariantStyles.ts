import { CustomTheme as Theme } from '../../../../types';
import {
  ButtonVariant,
  ContainedColors,
  OutlinedColors,
} from '../constants/types';

export const buttonVariantStyles = (
  variant: ButtonVariant,
  theme: Theme,
  color: ContainedColors | OutlinedColors
) =>
  ({
    outlined: `
      border: 1px solid;
      text-transform: uppercase;
        ${
          color === 'neutral'
            ? theme.palette.neutralDark_100
            : theme.palette[color]
        };

      &:disabled {
        color: ${
          color === 'neutral'
            ? theme.palette.neutralLight_100
            : theme.palette[color]
        };
        background-color: ${theme.colors.white};
        border-color: ${
          color === 'neutral'
            ? theme.palette.neutralLight_100
            : theme.palette[color]
        };
        cursor: default;
      }
    `,
    contained: `
      border: ${theme.spacing.none};
      text-transform: uppercase;
      &:disabled {
        opacity: 0.4;
        cursor: default;
      }
    `,
    text: `
      border: none;

      &:disabled {
        color: ${theme.palette.neutralLight_100};
      }
    `,
  }[variant]);
