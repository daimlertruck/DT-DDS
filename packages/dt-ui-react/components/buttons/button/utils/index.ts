export * from './buttonVariantColors';
export * from './buttonVariantStyles';
