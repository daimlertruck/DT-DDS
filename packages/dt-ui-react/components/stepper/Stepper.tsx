import { BaseProps, Orientation } from '../../types';

import { StepperStyled } from './Stepper.styled';

export interface StepperProps extends BaseProps {
  orientation?: Orientation;
}

const Stepper = ({
  children,
  dataTestId,
  style,
  orientation = 'vertical',
}: StepperProps) => {
  const testId = dataTestId ?? 'default';
  return (
    <StepperStyled
      data-testid={`${testId}-stepper`}
      orientation={orientation}
      style={style}
    >
      {children}
    </StepperStyled>
  );
};

export default Stepper;
