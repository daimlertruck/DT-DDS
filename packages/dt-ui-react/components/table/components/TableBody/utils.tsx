import { Children, ReactNode, cloneElement, isValidElement } from 'react';

import { TableDataCellHeader } from '../TableDataCell/TableDataCell';

export const addHeadersToDataCells = (
  dataCells: ReactNode,
  headers: ReactNode[]
) => {
  return Children.map(dataCells, (dataCell, index) => {
    if (isValidElement(dataCell)) {
      const header = headers[index];
      return cloneElement(
        dataCell,
        {},
        <>
          <TableDataCellHeader>{header}</TableDataCellHeader>
          {dataCell.props.children}
        </>
      );
    }
    return dataCell;
  });
};

export const addHeadersToDataCellsInRows = (
  rows: ReactNode,
  headers: ReactNode[]
) => {
  return Children.map(rows, (row) => {
    if (isValidElement(row)) {
      return cloneElement(
        row,
        {},
        addHeadersToDataCells(row.props.children, headers)
      );
    }
    return row;
  });
};
