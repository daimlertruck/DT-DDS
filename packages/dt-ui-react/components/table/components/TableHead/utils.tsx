import { Children, ReactNode, isValidElement } from 'react';

const extractColumnHeaders = (columnHeaders: ReactNode): ReactNode[] => {
  return Children.toArray(columnHeaders).reduce<ReactNode[]>(
    (headers, columnHeader) => {
      if (isValidElement(columnHeader)) {
        return [...headers, columnHeader.props.children];
      }

      return headers;
    },
    []
  );
};

export const extractTableHeaders = (theadRow: ReactNode) => {
  if (isValidElement(theadRow)) {
    return extractColumnHeaders(theadRow.props.children);
  }

  return [];
};
