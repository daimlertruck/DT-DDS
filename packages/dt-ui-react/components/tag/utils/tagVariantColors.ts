import { CustomTheme as Theme } from '../../../types';
import { TagColor, TagVariant } from '../constants';

interface TagVariantColors {
  border: string | number;
  color: string;
  borderColor?: string;
  backgroundColor: string;
  '&:hover'?: {
    borderColor?: string;
    backgroundColor: string;
    color: string;
  };
}

interface TagVariantColorsProps {
  theme: Theme;
  color: TagColor;
  variant: TagVariant;
  hasHover: boolean;
  isDisabled: boolean;
}

const getColorWithShade = ({
  theme,
  color,
  shade,
}: Pick<TagVariantColorsProps, 'color'> & {
  shade: '90' | '60' | '30' | '10';
  theme: Theme;
}) => {
  const { colors, palette } = theme;

  if (color === 'primary') {
    return {
      '90': palette.primaryDark_500,
      '60': palette.primaryDark_200,
      '30': palette.primaryLight_100,
      '10': palette.primaryLight_400,
    }[shade];
  }

  const colorWithShade: keyof Theme['colors'] = `${color}_${shade}`;

  return colors[colorWithShade];
};

export const tagVariantColors = ({
  theme,
  color,
  variant,
  hasHover,
  isDisabled,
}: TagVariantColorsProps): TagVariantColors => {
  const colorShade90 = getColorWithShade({
    theme,
    color,
    shade: '90',
  });
  const colorShade60 = getColorWithShade({
    theme,
    color,
    shade: '60',
  });
  const colorShade30 = getColorWithShade({
    theme,
    color,
    shade: '30',
  });
  const colorShade10 = getColorWithShade({
    theme,
    color,
    shade: '10',
  });

  const colorWithDisabled = isDisabled ? colorShade30 : colorShade60;

  return {
    solid: {
      border: theme.spacing.none,
      color: theme.colors.white,
      backgroundColor: colorWithDisabled,
      ...(hasHover && {
        '&:hover': {
          backgroundColor: colorShade90,
          color: theme.colors.white,
        },
      }),
    },
    outlined: {
      border: `1px solid ${colorShade60}`,
      color: colorWithDisabled,
      backgroundColor: colorShade10,
      borderColor: colorWithDisabled,
      ...(hasHover && {
        '&:hover': {
          borderColor: colorShade90,
          backgroundColor: colorShade10,
          color: colorShade90,
        },
      }),
    },
  }[variant];
};
