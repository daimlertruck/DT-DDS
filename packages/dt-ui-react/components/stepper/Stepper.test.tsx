import { render } from '@testing-library/react';
import React from 'react';

import { withProviders } from '../../utils';

import Stepper from './Stepper';

describe('<Stepper /> component', () => {
  const ProvidedStepper = withProviders(Stepper);

  describe('when the orientation is', () => {
    describe('vertical', () => {
      it('should render using flex direction column', () => {
        const { container } = render(
          <ProvidedStepper>
            <li>Item 1</li>
            <li>Item 2</li>
          </ProvidedStepper>
        );

        expect(container).toMatchSnapshot();
      });
    });

    describe('horizontal', () => {
      it('should render using flex direction row', () => {
        const { container } = render(
          <ProvidedStepper orientation='horizontal'>
            <li>Item 1</li>
            <li>Item 2</li>
          </ProvidedStepper>
        );

        expect(container).toMatchSnapshot();
      });
    });
  });
});
