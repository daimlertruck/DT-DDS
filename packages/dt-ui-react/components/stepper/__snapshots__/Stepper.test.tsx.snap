// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Stepper /> component when the orientation is horizontal should render using flex direction row 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 32px;
  list-style: none;
}

@media only screen and (min-width: 768px) {
  .emotion-0 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
  }
}

<div>
  <ul
    class="emotion-0 emotion-1"
    data-testid="default-stepper"
    orientation="horizontal"
  >
    <li>
      Item 1
    </li>
    <li>
      Item 2
    </li>
  </ul>
</div>
`;

exports[`<Stepper /> component when the orientation is vertical should render using flex direction column 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 32px;
  list-style: none;
}

@media only screen and (min-width: 768px) {
  .emotion-0 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
  }
}

<div>
  <ul
    class="emotion-0 emotion-1"
    data-testid="default-stepper"
    orientation="vertical"
  >
    <li>
      Item 1
    </li>
    <li>
      Item 2
    </li>
  </ul>
</div>
`;
